src/policy/sycl_policy_handler.cpp

```diff
@@ -57,15 +57,15 @@ namespace blas {
   template typename codeplay_policy::event_t                                   \
   PolicyHandler<codeplay_policy>::copy_to_device<element_t>(                   \
       const element_t *src, BufferIterator<element_t, codeplay_policy> dst,    \
-      size_t size = 0);                                                        \
```

Comment:
The idea here was not to force the user to pass the size. When the size is not passed it gets the size from the buffer and copies it. Otherwise, it uses the specific size passed by a user.

Reply:
The problem is that msvc doesn't seem to accept default parameters in template definitions like that. I will take another look

Reply:
It's fine to specify a default, but it cannot be done at the specialization or explicit instantiation. The best place to do it is in a forward declaration.

Reply:
@mcleary Any updates on this? If we can resolve the default parameters issue I think we are good to merge this.

Reply:
I'm looking into this. As soon as I have an update I'll let you know.

Reply:
@CodeplayBen @mehdi-goli I've added the default values for the size parameters in the actual function declaration so that MSVC can compile the code. Please take a look to see if that's ok.

I also updated the computecpp-sdk submodule because there are flags in FindComputeCpp.cmake required to build SYCL code on Windows with the latest Visual Studio.

Reply:
@mehdi-goli  This patch is working in Linux as well.

Reply:
Thanks @mcleary